Fix write_lobs "invalid byte sequence in UTF-8"

yegayazilim opened the pull request:
write_lobs checks to see whether the value is blank before executing the SQL statement. If the LOB is binary blank? call causes "invalid byte sequence in UTF-8" exception. We can instead use unless value, which is applicable both to strings and binary data.

yahonda commented:
Thank you for opening a pull request. It looks like the CI fails. Would you address this CI failure and add some RSpec test which makes sure what this pull request fixes.  Thanks.

yegayazilim commented:
I tried to write an RSpec test for the condition I'm fixing, but I couldn't. It happens when I try to save an .xlsx document to a binary column in Oracle.  write_lobs method uses value.blank? on this binary object and blank? throws "invalid byte sequence in UTF-8". So, I'm trying to avoid using blank? in the method. CI was failing in serialized column test case. In that test, value is serialized in write_lobs method. After serialization value can be nil because ActiveRecord persists empty Array and Hash as null in the database. So, I added another nil check after serialization to fix the CI problem. I also added a comment

yahonda commented:
CI for existing specs is green now. Would you add some specs to make sure which issue is fixed by this pull request? Thank you.

yahonda commented:
Closed via #2097
